compiler/src/iree/compiler/Dialect/Flow/Transforms/FusionUtils.cpp

```diff
@@ -56,6 +63,43 @@ static bool canInsOperandTieWithOutsOperand(OpOperand *insOperand) {
   return llvm::any_of(linalgOp.getOutputOperands(), canTieWithOutsOperand);
 }
 
+/// Checks if a linalg op is a simple reduction of the innermost dimensions
+/// with identity map for the input.
+static bool isSimpleReduction(linalg::LinalgOp linalgOp) {
+  SmallVector<Operation *, 4> combinerOps;
+
+  if (linalgOp.getNumOutputs() != 1) return false;
+
+  for (auto operand : linalgOp.getInputOperands()) {
+    auto inputIndexingMap = linalgOp.getTiedIndexingMap(operand);
+    if (!inputIndexingMap.isIdentity()) return false;
+  }
+
+  // Check if the result dims are d0, d1, ..., which means the reduction is done
+  // in the innermost dimensions.
+  auto output = linalgOp.getOutputOperand(0);
+  auto outputIndexingMap = linalgOp.getTiedIndexingMap(output);
+  for (const auto &en : llvm::enumerate(outputIndexingMap.getResults())) {
+    auto expr = en.value();
+    if (auto dim = expr.dyn_cast<AffineDimExpr>()) {
+      if (dim.getPosition() != en.index()) return false;
+    } else {
+      return false;
+    }
+  }
+
+  if (!matchReduction(linalgOp.getRegionOutputArgs(), 0, combinerOps) ||
+      combinerOps.size() != 1) {
+    return false;
+  }
+
+  return true;
+}
+
+static bool isBroadcast(AffineMap indexingMap) {
+  return !indexingMap.isIdentity() && indexingMap.isProjectedPermutation();
+}
```

Comment:
The name is a bit misleading as this doesn't really check for broadcast as a permutation would return true here. 
Do we want to support permutations?

Reply:
Updated.